osutil/mapper.go

```diff
@@ -0,0 +1,57 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package osutil
+
+import (
+	"fmt"
+	"os/exec"
+)
+
+// CreateLinearMapper creates a linear device mapping of the given device
+// with the given offset and size.
+//
+// The mapper device node is returned.
+func CreateLinearMapper(device, name, uuid string, offset, size uint64) (string, error) {
```

Comment:
nitpick, but this isn't the best name; when I first saw this PR I thought it was some sort of mathematical translational mapping when really it's a mapper device. I think something like `CreateLinearMapperDevice` would be better, at least something with MapperDevice in the name would be good I think